pyntc/devices/eos_device.py

```diff
@@ -187,25 +191,81 @@ def enable(self):
             self.native_ssh.exit_config_mode()
 
     @property
-    def facts(self):
-        if self._facts is None:
-            sh_version_output = self.show("show version")
-            self._facts = convert_dict_by_key(sh_version_output, BASIC_FACTS_KM)
-            self._facts["vendor"] = self.vendor
+    def vendor(self):
+        if self._vendor is None:
+            self._vendor = "arista"
+
+        return self._vendor
 
-            uptime = int(time.time() - sh_version_output["bootupTimestamp"])
-            self._facts["uptime"] = uptime
-            self._facts["uptime_string"] = self._uptime_to_string(uptime)
+    @property
+    def uptime(self):
+        if self._uptime is None:
+            self._uptime = self._get_uptime()
+
+        return self._uptime
+
+    @property
+    def uptime_string(self):
+        if self._uptime_string is None:
+            self._uptime_string = self._uptime_to_string(self._get_uptime())
 
+        return self._uptime_string
+
+    @property
+    def hostname(self):
+        if self._hostname is None:
             sh_hostname_output = self.show("show hostname")
-            self._facts.update(
-                convert_dict_by_key(sh_hostname_output, {}, fill_in=True, whitelist=["hostname", "fqdn"])
-            )
+            self._hostname = convert_dict_by_key(sh_hostname_output, {}, fill_in=True, whitelist=["hostname"])[
+                "hostname"
+            ]
+
+        return self._hostname
 
-            self._facts["interfaces"] = self._get_interface_list()
-            self._facts["vlans"] = self._get_vlan_list()
+    @property
+    def interfaces(self):
+        if self._interfaces is None:
+            self._interfaces = self._get_interface_list()
+
+        return self._interfaces
+
+    @property
+    def vlans(self):
+        if self._vlans is None:
+            self._vlans = self._get_vlan_list()
```

Comment:
lets move this logic from the internal method to hear and whatever needs that can just reference the property